marlowe/test/test-report.md

```diff
@@ -0,0 +1,543 @@
+- The [On-Chain Transaction Specification for Marlowe](../specification/ReadMe.md) defines the realization of Marlowe on the Cardano blockchain. In particular, this specification relates abstract Marlowe types and constructs to the concrete types present in Cardano's Plutus language in such a way the Marlowe transactions conform to Cardano's [Ledger Specifications](https://github.com/input-output-hk/cardano-ledger#cardano-ledger) and the [Formal Specification of the Plutus Core Language](https://github.com/input-output-hk/plutus/blob/master/plutus-core-spec/draft-new-specification/plutus-core-specification.pdf).
+
+The bridge between abstract semantics and concrete Plutus is the [`Language.Marlowe.Core.V1.Semantics`](../src/Language/Marlowe/Core/V1/Semantics.hs) function `computeTransaction`:
+```haskell
+computeTransaction :: TransactionInput -> State -> Contract -> TransactionOutput
+```
+where the blockchain-agnostic types  `TransactionInput` , `State`, `Contract`, and `TransactionOutput` are the input to the contract, its pre-transaction state, its pre-transaction contract, and the result of applying the input to the contract. Note that applying input to a Marlowe contract results in a new, smaller contract that is the sub-contract of the original pre-transaction contract: essentially, a Marlowe contract consists of a directed acyclic graph (DAG) of contract continuations, where branching is determined by applying inputs or experiencing timeouts. This DAG-continuation has advantages not only for proving guarantees on Marlowe semantics, but also for minimizing the on-chain storage needed for a contract.
+
+Marlowe transactions utilize one of two Plutus V2 validators, residing in the [`Language.Marlowe.Scripts`](../src/Language/Marlowe/Scripts.hs) module:
+- The *Semantics Validator* is Plutus code that verifies that a transaction obeys the Marlowe DSL semantics along with the Cardano ledger rules. This validator resides in the `smallUntypedValidator` function that marshals Plutus types to call `computeTransaction` and checks constraints on input and output.
+- The *Payout Validator* is Plutus code that verifies that a transaction is properly authorized by a Marlowe role to withdraw funds paid by the Semantics Validator. This validator resides in the  `rolePayoutValidator`  function that simply verifies authorization.
+
+Marlowe's property-based and golden tests reside in the `marlowe-test` executable of the [marlowe](../) package of the [marlowe-cardano](https://github.com/input-output-hk/marlowe-cardano/) repository, while Marlowe's on-chain tests reside in the [`run-nonpab-test.sh`](../../marlowe-cli/run-nonpab-tests.sh) script of the [marlowe-cli](../../marlowe-cli/) package of the repository. The test suite `marlowe-actus-test` in the [marlowe-actus](../../marlowe-actus/) package heavily stresses the Haskell implementation of Marlowe semantics, too. Informal tests are scattered throughout the Marlowe repositories.
```

Comment:
Suggesting to remove the comment about ACTUS tests here. When separating the ACTUS code base into ACTUS core (no dependencies to Marlowe, resp. Cardano) and the Marlowe specific part, I put the ACTUS test-framework into ACTUS core only.
```suggestion
Marlowe's property-based and golden tests reside in the `marlowe-test` executable of the [marlowe](../) package of the [marlowe-cardano](https://github.com/input-output-hk/marlowe-cardano/) repository, while Marlowe's on-chain tests reside in the [`run-nonpab-test.sh`](../../marlowe-cli/run-nonpab-tests.sh) script of the [marlowe-cli](../../marlowe-cli/) package of the repository. Informal tests are scattered throughout the Marlowe repositories.
```